### ISSUE TITLE
RFE: Please document how the iOS virtual is configured (eg getting a Done button or not)

### ISSUE THREAD

**Issue by ThomasH99** [codenameone/CodenameOne#3126]
As can be seen in the discussion here https://stackoverflow.com/questions/61031441/how-to-close-a-textfield-numeric-keyboard-on-ios-after-editing, it is currently not documented what the principles for configuring the virtual iOS keyboard are. When is a Next button added, when is no button added (only carriage return) and when is a Done button added. 

Personally, I've spend many, many hours trying to get this to work (at first I thought my code was wrong), and since it is a critical part of a good user experience it really should be made easy to get it right, and just a few lines of documentation describing how it works will help your users save time. My apologies if it is already documented somewhere, but I tried searching with no success. 

I'm currently only testing on iOS, so not sure if the RFE is also relevant for Android.

**Comment by ThomasH99** [comment]
I would really appreciate if you had the opportunity to clarify this, I've struggled with it and spend a lot of time experimenting, but not really being able to build an understanding. And it really makes the user experience suck not being able to ensure there's a Done button in the keyboard to close it when needed. My guess is that a few words of documentation from you could help clarify this authoritatively :-)

**Comment by shannah** [comment]
Generally the "Done" button is added when it is the last field on the form that can be "tabbed" to.
A "Next" button is added when it is not the last field on the form that can be "tabbed" to.  

In some cases (when it is just numeric input) it may have both a "Done" and a "Next" button.

You can set the "iosHideToolbar" client property on the TextField to "true" to hide the toolbar altogether.

The logic is a bit complicated so there may be other edge cases.

**Comment by ThomasH99** [comment]
Thanks Steve, 

This makes sense, and it is consistent with a simple test case (both my own simple test example, and also like @jsfan3 mentioned on StackOverflow). 

But in my app this is *not* happening, which I guess it should if your description is exhaustive? So, it seems there is something else influencing whether the buttons appear or not. Is your code implementing this available somewhere? Then I will try to see if I can understand what might be happening in my app.

**Comment by ThomasH99** [comment]
Any additional help possible on this one? It has a major impact on the UI and I just can't figure out what is going on.

**Comment by jsfan3** [comment]
I suppose, in cases like this, the only way to really ask for help and to be able to give it is to offer a minimal test case, which will always reproduce the problem.

**Comment by ThomasH99** [comment]
Crap, I think I finally found the reason. I had left the code `Display.getInstance().setProperty("iosHideToolbar", "true");` in the start() of the app. Probably I added that a long time ago to not forget that the option existed after I read about it in this discussion https://stackoverflow.com/questions/48727116/codename-one-done-button-of-ios-virtual-keyboard. I haven't tested on the device yet, but pretty confident this was the reason. Thanks for everyone's help!  (maybe worthwhile adding this option to the documentation if not already the case)
